src/Spout/Writer/XLSX/Internal/Worksheet.php

```diff
@@ -133,10 +133,18 @@ public function addRow($dataRow, $style)
 
         $rowXML = '<row r="' . $rowIndex . '" spans="1:' . $numCells . '">';
 
-        foreach($dataRow as $cellValue) {
+        foreach($dataRow as $cell) {
+            if (is_array($cell)) {
```

Comment:
Why shouldn't a value be able to be generated with multiple types (I agree, would be nicer with a class instead of whats happening now). Since a value and its interaction within the sheet can be different depending on type, especially when running around with different formats